jbpm-flow/src/main/java/org/jbpm/workflow/instance/node/MilestoneNodeInstance.java

```diff
@@ -104,7 +104,7 @@ public void matchCreated(MatchCreatedEvent event) {
             // trigger node instances of that milestone node
             String ruleName = event.getMatch().getRule().getName();
             String milestoneName = "RuleFlow-Milestone-" + getProcessInstance().getProcessId() + "-" + getMilestoneNode().getUniqueId();
-            if (milestoneName.equals(ruleName) && checkProcessInstance((Activation) event.getMatch()) && checkDeclarationMatch(event.getMatch(), getMilestoneNode().getMatchVariable())) {
+            if (((Activation) event.getMatch()).isActive() &&  milestoneName.equals(ruleName) && checkProcessInstance((Activation) event.getMatch()) && checkDeclarationMatch(event.getMatch(), getMilestoneNode().getMatchVariable())) {
```

Comment:
Can you explain briefly what was going on in that case with a subprocess from JIRA? I mean when there was a match for the rule at the same time for 2 objects, and the second match should be cancelled because we need only one (we configured it in that way), does it mean that drools will fire a MatchCreatedEvent also for the second rule, but this time with the match set as not active?

Reply:
that's exactly my understanding as well so you actually explained it yourself :)